back/engines/free/user_confirmation/app/interactors/user_confirmation/send_confirmation_code_to_user.rb

```diff
@@ -0,0 +1,14 @@
+module UserConfirmation
+  class SendConfirmationCodeToUser < ApplicationInteractor
+    delegate :user, to: :context
+
+    def call
+      unless user.registered_with_email?
+        fail_with_error! :registration_method, :invalid, message: 'Confirmation is currently working for emails only.'
+      end
+
+      ConfirmationsMailer.with(user: user).send_confirmation_code.deliver_later
+      user.update(email_confirmation_code_sent_at: Time.zone.now)
```

Comment:
I'm also slightly concerned about silent update failures here.

Reply:
The only solution I see here is to change to `deliver_now` instead. Do you think it would be acceptable?

Reply:
I don't have a strong preference between `deliver_now` and `deliver_later` in this case, as long as it doesn't take too long for the email to be delivered (as I understand this will currently block the endpoint from returning a response?). 

I guess that means Queue won't retry sending out the email if there's a failure? But I guess Mailgun already handles retrying email delivery for us, so it's probably not an issue.